plugins/modules/cloud/misc/proxmox_snap.py

```diff
@@ -137,14 +147,31 @@ def snapshot_remove(self, vm, vmid, timeout, snapname, force):
             time.sleep(1)
         return False
 
+    def snapshot_rollback(self, vm, vmid, timeout, snapname):
+        if self.module.check_mode:
+            return True
+
+        taskid = self.snapshot(vm, vmid)(snapname).post("rollback")
+        while timeout:
+            if (self.proxmox_api.nodes(vm['node']).tasks(taskid).status.get()['status'] == 'stopped' and
+                    self.proxmox_api.nodes(vm['node']).tasks(taskid).status.get()['exitstatus'] == 'OK'):
+                return True
+            timeout -= 1
+            if timeout == 0:
```

Comment:
Assume `timeout == 1`. Then it will check the condition, and if it's not satisfied will immediately fail instead of waiting for a second. If you change this to `< 0`, then it will wait one second and check the conditions again. I think this is more closely to what `timeout` should be.
```suggestion
            if timeout < 0:
```

Reply:
I have moved the subtraction instead for better readability.
The two changes accidentally got into one commit here, but since the PR will be squashed this should be no problem, right?

Reply:
Yes, that's no problem.